### PR TITLE
[cli] Batch: support multiple/nested !include directive

### PR DESCRIPTION
**Pull request by dtiller**
**Issue 7311**
This change updates the '!include' directive available in the 'batch' CLI command to handle multiple includes as well as nested includes.

Instead of parsing the files into a tree, parsing them via the streaming API allows multiple and nested includes. 

The appropriate test case (GenerateBatchTest) has been updated to test nested and multiple includes for both JSON and YAML.

The file usage.md has been updated to explain the change. It mentions version 5.0.0 as the affected version - that may have to change.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [ ] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by jimschubert** [comment]
I did a quick scan and this looks really good. I'll review more in depth later this week.

It doesn't look like your commits are associated with your GitHub profile, so you wouldn't display as a contributor after merge unless you've associated the email used in the commits with your profile. Let me know if you'd need help resolving this.

Also cc'ing core team since we don't have an explicitly defined CLI review team. @OpenAPITools/generator-core-team

**Comment by dtiller** [comment]
Jim,

Thanks for the kind words - I've only recently started using the generator, and it's a very useful tool. I'm particularly happy to see the work you all have put in to support customization.

FYI, I also live in Richmond (actually Hanover county)! Small world. 

I turned on 'keep my email private' and changed my local git email to the private one they provide - 3858971+dtiller@users.noreply.github.com <mailto:3858971+dtiller@users.noreply.github.com>. Do I need to do a push to get github to see the connection? If so I'll go ahead and change the documentation a little and push. I don't want to push unnecessarily and trigger another round of CI tests!

Thanks for the review!

David Tiller

File: docs/usage.md [hunk 1]
```diff
@@ -602,7 +602,7 @@ openapi-generator generate -i petstore.yaml -g typescript-fetch -o out \
 The `batch` command allows you to move all CLI arguments supported by the `generate` command into a YAML or JSON file.
 
 *NOTE*: This command supports an additional `!include` property which may point to another "shared" file, the base path to which can be
-modified by `--includes-base-dir`.
+modified by `--includes-base-dir`. Starting with 5.0.0, the `!batch` command supports multiple `!include` properties, either sequential or nested. In order to support multiple `!include` properties in a JSON file, the property name can have a suffix, e.g. `!include1`, `!include2`, etc. The suffix have no meaning other than providing unique property names.
```
Comment (jimschubert) [inline docs/usage.md:605]: Should be `batch` not `!batch` in the first changed sentence.
And `The suffixes` in the last sentence.

```suggestion
modified by `--includes-base-dir`. Starting with 5.0.0, the `batch` command supports multiple `!include` properties, either sequential or nested. In order to support multiple `!include` properties in a JSON file, the property name can have a suffix, e.g. `!include1`, `!include2`, `!include_asdf`, etc. The suffixes have no meaning other than providing unique property names. The only requirement is that the property start with `!include`. Nested documents are loaded and merged in declaration order.
```

File: modules/openapi-generator-cli/src/test/resources/batch/jaxrs-datelib-j8-nested-include.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+"!include": common/jaxrs-datelib-j8.yaml
+generatorName: jaxrs-jersey
+inputSpec: batch/specs/petstore.yaml
```
Comment (jimschubert) [inline modules/openapi-generator-cli/src/test/resources/batch/jaxrs-datelib-j8-nested-include.yaml:4]: I wonder if we should move `inputSpec` from here to `jaxrs-datelib-j8-nested.yaml` to verify that the initial `!include` processing will also load sibling properties?

File: modules/openapi-generator-cli/src/test/resources/batch/jaxrs-datelib-j8-nested-include.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "!include": "common/jaxrs-datelib-j8.json",
+  "generatorName": "jaxrs-jersey",
+  "inputSpec": "batch/specs/petstore.yaml",
```
Comment (jimschubert) [inline modules/openapi-generator-cli/src/test/resources/batch/jaxrs-datelib-j8-nested-include.json:4]: I wonder if we should move `inputSpec` from here to `jaxrs-datelib-j8-nested.json` to verify that the initial `!include` processing will also load sibling properties?

I tested the suggestion locally for both JSON and YAML to verify the functionality.

**Review by jimschubert — COMMENTED** [review]
Looks good! I just made a few small suggestions. I won't merge until you've updated your email in the commits.

**Comment by jimschubert** [comment]
@dtiller Small world indeed! I'm actually on the south side, about 10 minutes from the zoo.

For the email, you'll probably want to do an interactive rebase against master and edit each commit to modify the author information, then force push to your branch. Git Tower has a pretty good walkthrough here: [Using Interactive Rebase](https://www.git-tower.com/learn/git/faq/change-author-name-email#chapter_using+interactive+rebase).

**Comment by dtiller** [comment]
Jim,

I attempted the rebase and forced push, but now the CI tests are failing horribly with conflicts:

Conflicting files
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoClientCodegen.java
modules/openapi-generator/src/main/resources/cpp-ue4/api-source.mustache
modules/openapi-generator/src/main/resources/go/api.mustache
samples/client/petstore/go/go-petstore/api_another_fake.go
samples/client/petstore/go/go-petstore/api_fake.go

I think I dorked it up royally. What do you suggest? I can certainly nuke the branch from orbit and start over - there are only a couple of files with changes.

**Comment by jimschubert** [comment]
@dtiller no worries, I can fix all that up. The main piece was getting author information correct on the commits.

**Pull request merged**